spec/lib/pedicel/base_spec.rb

```diff
@@ -0,0 +1,425 @@
+    #end
+    #
+    # Intentionally removed. It would fail with the current implementation.
+    #
+    # The thought behind the test is that if root was used in all three places,
+    # then all conditions are fulfilled, but it would also be quite strange to
+    # see Apple's Root CA certificate sign a payment token. Thus, we accept that
+    # this theoretically acceptable chain is not accepted (because it will never
+    # happen).
+
+    it 'errs if intermediate is not signed by root' do
+      params[:root] = PedicelPay::Backend.generate.ca_certificate
+
+      expect{Pedicel::Base.verify_x509_chain(params)}.to raise_error(Pedicel::SignatureError, /invalid chain/)
+    end
+
+    it 'errs if leaf is not signed by intermediate (1)' do
+      params[:leaf] = PedicelPay::Backend.generate.leaf_certificate
+
+      expect{Pedicel::Base.verify_x509_chain(params)}.to raise_error(Pedicel::SignatureError, /invalid chain/)
+    end
+
+    it 'errs if leaf is not signed by intermediate (2)' do
+      another_backend = PedicelPay::Backend.generate
+      params[:root]         = another_backend.ca_certificate
+      params[:intermediate] = another_backend.intermediate_certificate
+
+      expect{Pedicel::Base.verify_x509_chain(params)}.to raise_error(Pedicel::SignatureError, /invalid chain/)
+    end
+
+    it 'errs if leaf is not signed by intermediate and intermediate is not signed by root' do
+      params[:root]         = PedicelPay::Backend.generate.ca_certificate
+      params[:intermediate] = PedicelPay::Backend.generate.intermediate_certificate
+
+      expect{Pedicel::Base.verify_x509_chain(params)}.to raise_error(Pedicel::SignatureError, /invalid chain/)
+    end
+
+    it 'is truthy when the chain is good' do
+      expect(Pedicel::Base.verify_x509_chain(params)).to be_truthy
+    end
+  end
+
+
+  describe 'Pedicel::Base.verify_signed_time' do
+    let (:signature) { OpenSSL::PKCS7.new(pedicel.signature) }
+    let (:now) { signature.signers.first.signed_time }
+
+    it 'does not err when all checks are good' do
+      expect{Pedicel::Base.verify_signed_time(signature: signature, now: now)}.to_not raise_error
+    end
+
+    let (:limit) { Pedicel.config[:replay_threshold_seconds] }
+
+    it 'errs if the signature is too new' do
+      expect{Pedicel::Base.verify_signed_time(signature: signature, now: now-limit)}.to_not raise_error
+      expect{Pedicel::Base.verify_signed_time(signature: signature, now: now-limit-1)}.to raise_error(Pedicel::SignatureError)
+    end
+
+    it 'errs if the signature is too old' do
+      expect{Pedicel::Base.verify_signed_time(signature: signature, now: now+limit)}.to_not raise_error
+      expect{Pedicel::Base.verify_signed_time(signature: signature, now: now+limit+1)}.to raise_error(Pedicel::SignatureError)
+    end
+
+    it 'is truthy when all checks are good' do
+      expect(Pedicel::Base.verify_signed_time(signature: signature, now: now)).to be_truthy
```

Comment:
It many other tests the text has been `it '... is truthy ...'`, however, with an `expect ... .to be true`. Here you are actually using `be_truthy`. Perhaps tighten it to `be true`, as you are in complete control of the method's return values?

Reply:
`Base#verify_signature` returns `self` (to allow method chaining) and is therefore also expected to `be_truthy`, but in this case this would never make sense. The only other truthy (but not `true`) return value I could think of from `Base.verify_signed_time` would be the signing time, but that's probably a bit far fetched.

Thus, I agree. Same applies for `Base.verify_root_certificate` and `Base.verifyx509_chain` btw. I'll fix both.